src/components/OfflineIndicator.tsx

```diff
@@ -46,7 +48,25 @@ function OfflineIndicator({style, containerStyles}: OfflineIndicatorProps) {
                 width={variables.iconSizeSmall}
                 height={variables.iconSizeSmall}
             />
-            <Text style={[styles.ml3, styles.chatItemComposeSecondaryRowSubText]}>{translate('common.youAppearToBeOffline')}</Text>
+            <Text style={[styles.ml3, styles.chatItemComposeSecondaryRowSubText]}>
+                {
+                    // If we reversed the ternary, unreachability message would always show even when offline
```

Comment:
I still don't like this comment. Unless you feel very strongly about it, I would just remove it. It's **normal** that you can't just flip the ternary expression and expect it to work.

If I really wanted to make this logic more clear, I'd consider extracting the whole markup to a separate component, like `BaseOfflineIndicator`, which accepts children.

Then, we could update the `return` logic of the `OfflineIndicator`:

```tsx
if (isOffline) {
  return <BaseOfflineIndicator>{translate('common.youAppearToBeOffline')}</BaseOfflineIndicator>
} else if (!isBackendReachable) {
  return (
    <BaseOfflineIndicator>
        {translate('common.weMightHaveProblem')}
        <TextLink
            href={CONST.STATUS_EXPENSIFY_URL}
            style={[styles.chatItemComposeSecondaryRowSubText, styles.link]}
        >
            {new URL(CONST.STATUS_EXPENSIFY_URL).host}
        </TextLink>
        .
    </BaseOfflineIndicator>
  );
} else {
  return null;
}
```

This is clear without comments.

I'll leave it to you: either just remove the comment, or (if you like the idea) you can also do the suggested refactoring to make the logic clear without any comments.